### DISCUSSION TITLE
display hints for parameter names hint for CLI mutable constructors (dotnet constructors) and new constrcutor (.net classes)

### DISCUSSION THREAD

**Discussion by jkone27** [ionide/ionide-vscode-fsharp#1948] [Ideas]
would be nice to have instead of this 
<img width="722" alt="image" src="https://github.com/ionide/ionide-vscode-fsharp/assets/10419217/054057f7-8d7d-473b-89e7-0c07d817992a">

something like this, a bit like in rider, to show user the name of parameters, can help when multiple parameters have the same type like in many nuget packages or .net / C#  libraries, to not swap them by mistake!

<img width="725" alt="image" src="https://github.com/ionide/ionide-vscode-fsharp/assets/10419217/cf154282-f3f7-4e9e-92b7-903f44ffffd9">

**Comment by baronfel** [comment]
I think this is something we can do, but we were waiting for a fix in the FCS APIs to be released. It would be worth checking again, I think there's some commented out code in our current inlay hints code that points to this gap.

**Reply by baronfel** [reply]
I knew we could do it :)

![image](https://github.com/ionide/ionide-vscode-fsharp/assets/573979/3f530869-2cb3-4607-b38b-f293d5a12c18)

**Reply by baronfel** [reply]
https://github.com/fsharp/FsAutoComplete/pull/1176 I'll merge this later today and push it in a release this weekend. Thanks for the poke, I'd been meaning to do this :)

**Reply by jkone27** [reply]
this is **FREAKIN** awesome!

**Reply by baronfel** [reply]
This is released on Ionide 7.15!
